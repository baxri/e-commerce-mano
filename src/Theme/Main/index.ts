import button from "./button";
import colors from "./colors";
import common from "./common";
import fonts from "./fonts";

export default {
  name: "Main Theme",
  dark: false,
  button,
  common,
  colors,
  fonts,

  // --   0  1  2  3  4   5   6   7   8   9
  space: [0, 2, 4, 8, 12, 16, 20, 24, 32, 64],
  // --       0   1   2   3    4   5   7   8   9
  fontSizes: [10, 12, 14, 16, 18, 20, 22, 24, 26],
  // --         0   1   2   3    4   5   7   8   9
  lineHeights: [13, 15, 17, 19, 21, 23, 25, 27, 29],
  // --         0    1    2    3
  fontWeights: [400, 500, 600, 700],
};
