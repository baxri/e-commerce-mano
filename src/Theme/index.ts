import MainTheme from "./Main";

enum ThemeID {
  Main = "main",
}

const THEMES: Record<ThemeID, any> = {
  [ThemeID.Main]: MainTheme,
};

export { ThemeID, THEMES };
