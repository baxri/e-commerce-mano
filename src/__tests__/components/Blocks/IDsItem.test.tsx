import React from 'react';
import PersonalIcon from 'assets/icons/dashboard/personal.svg';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { ThemeID, THEMES } from 'Theme';

import IDsItem from 'Components/Blocks/IDsItem';

test('IDsItem', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={THEMES[ThemeID.Main]}>
        <IDsItem icon={<PersonalIcon />} title="Personal info" />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
