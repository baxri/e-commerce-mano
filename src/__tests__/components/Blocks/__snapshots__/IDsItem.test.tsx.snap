// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDsItem 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FAFAFB",
      "borderRadius": 16,
      "flexDirection": "row",
      "flexGrow": 1,
      "height": 80,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    alignCenter={true}
    stretch={true}
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      color="black"
      fontFamily="Inter"
      fontSize={2}
      fontWeight={0}
      ml={5}
      style={
        [
          {
            "color": "#0F1120",
            "fontFamily": "Inter",
            "fontSize": 14,
            "fontWeight": "400",
            "marginLeft": 16,
          },
        ]
      }
    >
      Personal info
    </Text>
  </View>
  <View
    alignEnd={true}
    stretch={true}
    style={
      [
        {
          "alignItems": "flex-end",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
